### PR TITLE
Gamepad Extensions for Kotlin

### PR DESCRIPTION
**Pull request by JIceberg**
# Pull Requests

Please note that we accept pull requests from anyone, but that does not mean it will be merged.

## What kind of change does this PR introduce?
* Feature - adds gamepad extensions in Kotlin

## Did this PR introduce a breaking change?
* No

__Please make sure your PR satisfies the requirements of [the contributing page](CONTRIBUTING.md)__

### CODE REVIEW

**Review by puravdatta-sudo — APPROVED** [review]
Looks good to me, not too many changes I see?

**Comment by JIceberg** [comment]
@puravdatta-sudo bro it was a draft PR :omegalul:

**Comment by JarnaChao09** [comment]
> Looks good to me, not too many changes I see?

someone didn't see the TODO

File: core/src/main/java/com/arcrobotics/ftclib/controller/PIDFController.java [hunk 1]
```diff
@@ -110,6 +110,8 @@ public double getSetPoint() {
      */
     public void setSetPoint(double sp) {
```
Comment (Lunerwalker2) [inline core/src/main/java/com/arcrobotics/ftclib/controller/PIDFController.java:111]: Should we add a set point for velocity in this as well? And, relating to that, maybe also an acceleration setpoint?

File: core/src/main/java/com/arcrobotics/ftclib/gamepad/TriggerReader.java [hunk 2]
```diff
@@ -1,74 +1,66 @@
...
-        }
-        lastState = currState;
-        triggerName = "";
-    }
 
-    public TriggerReader(GamepadEx gamepad, GamepadKeys.Trigger trigger, String triggerName, Telemetry telemetry) {
-        this.gamepad = gamepad;
-        this.trigger = trigger;
-        if(this.gamepad.getTrigger(trigger) > 0.5) {
+        if (this.gamepad.getTrigger(trigger) > 0.5) {
```
Comment (Lunerwalker2) [inline core/src/main/java/com/arcrobotics/ftclib/gamepad/TriggerReader.java:23]: `currState = this.gamepad.getTrigger(trigger) > 0.5;`

**Review by Lunerwalker2 — COMMENTED** [review]
Everything looks good, although some code can be simplified and maybe some more options added for the PID controller.

**Comment by JIceberg** [comment]
Let's squash and merge this after it gets reviewed.

**Review by JIceberg — COMMENTED** [review]
So this is based on the `Trigger` class, right? Where/what exactly is the gamepad extension purpose here?

**Comment by JarnaChao09** [comment]
> So this is based on the `Trigger` class, right? Where/what exactly is the gamepad extension purpose here?

It was originally to make working with the GamepadButtons easier in command based in Kotlin. However, upon looking at the hierarchy, I found that all of the methods used are in the Trigger class, so if we ever added another type of button separate from GamepadButton, it would be very easy to use these extensions with very few changes to the source code.

**Comment by JIceberg** [comment]
Do you have a sample implementation of the gamepad with these extensions?

**Comment by JarnaChao09** [comment]
> Do you have a sample implementation of the gamepad with these extensions?

I can add on really quickly based on the GamepadSample.java

File: examples/src/main/java/com/example/ftclibexamples/kotlinexamples/InterpLUTExtSample.kt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import com.arcrobotics.ftclib.kotlin.extensions.util.toLUT
+import com.arcrobotics.ftclib.util.LUT
+import com.qualcomm.robotcore.eventloop.opmode.Disabled
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp
+
+
+@TeleOp(name = "InterpLUTExt Sample")
+@Disabled
+class InterpLUTExtSample: LinearOpMode() {
```
Comment (puravdatta-sudo) [inline examples/src/main/java/com/example/ftclibexamples/kotlinexamples/InterpLUTExtSample.kt:18]: name change

**Review by puravdatta-sudo — APPROVED** [review]
I like the samples.

**Pull request merged**